pex/third_party/__init__.py

```diff
@@ -100,9 +100,12 @@ def containing(cls, root):
     prefix = ''
     path = root
     while path:
+      # We replace os.sep with '/' here because the zip file format spec specifies that paths must
+      # use forward slashes. See section 4.4.17 of
+      # https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT.
       if zipfile.is_zipfile(path):
-        return cls(zipfile_path=path, prefix=prefix + os.sep if prefix else '')
-      prefix = os.path.join(prefix, os.path.basename(path))
+        return cls(zipfile_path=path, prefix='{}{}'.format(prefix, os.sep).replace(os.sep, '/'))
+      prefix = os.path.join(prefix, os.path.basename(path)).replace(os.sep, '/')
```

Comment:
os.path.join is os.sep aware and roughly the whole point of this PR is os.sep sensitivity was wrong. Since this loop builds up a prefix from path components not containing os.sep by construction (os.path.basename ensures this), why not be more direct and (I reversed the joining as well here per your discovery of that bug):
```python
prefix = '{}/{}'.format(os.path.basename(path), prefix)
```

Reply:
Fixed as well, although I needed a guard for the base case since otherwise it will become `basename/`, which breaks later when we append another `/`.